### DISCUSSION TITLE
是否应该提供为修改已注册的Matcher的方法

### DISCUSSION THREAD

**Discussion by abrahum** [nonebot/discussions#20] [通用]
nb2现在已经有很多插件可以非常便利的实现各种方法。
但是现有插件用户似乎缺少一种便利地给由插件注册的各种`Matcher`增删`rule`、`permission`等`Matcher`属性的方法。
目前想要修改似乎只能直接操作`matchers`？
是否考虑为这个功能提供一些包装好的方法呢？
~~（或者额外写个插件提供这种功能扩展）~~

**Comment by yanyongyu** [comment]
`Matcher` 可以直接从插件导入后修改priority属性等，不过由于rule permission都是函数不太好判断删除哪个
